### PR TITLE
Initialize the recording stack upon init of Problem

### PR DESCRIPTION
**Pull request by hschilling**

### CODE REVIEW

File: openmdao/core/problem.py [hunk 1]
```diff
@@ -103,6 +103,8 @@ def __init__(self, model=None, comm=None, use_ref_vector=True, root=None):
 
         self._mode = None  # mode is assigned in setup()
 
+        recording_iteration_stack = []
+
```
Comment (Kenneth-T-Moore) [inline openmdao/core/problem.py]: So, I did try this when I found the test failure problem, and it didn't fix it for me. Do you maybe have access to a linux system that you can test it on?

Comment (hschilling) [inline openmdao/core/problem.py]: I could try it on the OpenMDAO cluster, right? Will take me a while to setup the environment there.

Comment (kmarsteller) [inline openmdao/core/problem.py]: @hschilling : see discussion in #case-recorders

Comment (hschilling) [inline openmdao/core/problem.py]: So I assume I should hold off on this since it's not crucial?

Comment (Kenneth-T-Moore) [inline openmdao/core/problem.py]: No, just merge it. I think there is a story too.

**Pull request merged**